### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'auto')

### ISSUE THREAD

**Issue by NickVanzo** [sc-forks/solidity-coverage#785]
Hi, I'm trying to run solidity coverage in my Hardhat project but everytime I try to run the command:
```
npx hardhat coverage --testfiles "./test/*.test.ts" --show-stack-traces
```
I get this error in the console:
```
Compilation:
============

Nothing to compile
No need to generate any newer typings.
Error in plugin solidity-coverage: TypeError: Cannot read properties of undefined (reading 'auto')

HardhatPluginError: TypeError: Cannot read properties of undefined (reading 'auto')
    at SimpleTaskDefinition.action (/Users/nicolovanzo/Documents/work/Circolo/contracts/node_modules/solidity-coverage/plugins/hardhat.plugin.js:274:35)
    at async Environment._runTaskDefinition (/Users/nicolovanzo/Documents/work/Circolo/contracts/node_modules/hardhat/src/internal/core/runtime-environment.ts:330:14)
    at async Environment.run (/Users/nicolovanzo/Documents/work/Circolo/contracts/node_modules/hardhat/src/internal/core/runtime-environment.ts:163:14)
    at async main (/Users/nicolovanzo/Documents/work/Circolo/contracts/node_modules/hardhat/src/internal/cli/cli.ts:277:7)
```

I don't know why this is happening. Anyone?

**Comment by cgewecke** [comment]
Is this still happening?

**Comment by cgewecke** [comment]
@clauBv23 Would it be possible to link to a way of reproducing this error so I can try to debug?

**Comment by clauBv23** [comment]
@cgewecke Thanks for your fast response. 
I did a debug and found the issue was happening when creating [the provider for the network](https://github.com/sc-forks/solidity-coverage/blob/bfb7e322253a4ddfdb8c09bbbc9166c453f19400/plugins/resources/nomiclabs.utils.js#L72-L85). Then checked the Hardhat code and realized that there were some problems with the network configuration that was receiving. 
Indeed, I changed the network configuration because needed to predefine some accounts and there were some missing parameters. 

So the 
`TypeError: Cannot read properties of undefined (reading 'auto')` error was due to not having configured:
 ```
mining: {
        auto: true,
        interval: 5000,
      },
```
 when trying to access `networkConfig.mining.auto`. 

I updated the configuration for the correct one and it worked like a charm 🙌
